### DISCUSSION TITLE
Integration with VS Code

### DISCUSSION THREAD

**Discussion by geekox86** [modernweb-dev/web#1246] [Q&A]
Thank you for this great test runner.
Is there an existing or planned VS Code extension to better integrate WTR with VS Code for running and debugging tests as well as reporting in VS Code problems window?

**Comment by daKmoR** [comment]
We would love to have better integration with VS Code... however so far it's just wishful thinking.

As far as I can tell so far no one investigated how much work it would be to set it up 😅  

we would certainly welcome a contribution in this direction 🤗

**Comment by Manish3323** [comment]
I have given it a try.. 
our team's been using this extension in vscode for more than a week now..
[vs-code @web/test-runner extension](https://marketplace.visualstudio.com/items?itemName=manishgowardipe.test-runner)

**Comment by stagas** [comment]
I've also given it a try: 

- [VSCode extension](https://marketplace.visualstudio.com/items?itemName=stagas.vscode-web-test-runner-explorer)
- [Repo](https://github.com/stagas/vscode-web-test-runner-explorer)

It integrates with [Test Explorer UI](https://marketplace.visualstudio.com/items?itemName=hbenl.vscode-test-explorer), very limited atm but it should be a starting point for further work.

**Comment by melink14** [comment]
I think the most useful integration point is being able to set breakpoints while running tests for quick debugging and interation. I had this working when using the snowpack plugin usinga normal attach debugger profile and watch mode but am having this trouble using just esbuild. 

The debugger attaches but when a breakpoint is hit, the test stops running but no visible changes occur in VS Code.

Will debug more later; it may be an issue with WTR's esbuild config...
